### PR TITLE
Add AWS naming convention for IAM roles

### PR DESCRIPTION
**Pull request by pipo02mix**

### CODE REVIEW

File: aws/naming_resources.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Naming Resources
+
+## IAM Resources
+
+AWS operator needs a set of roles in customer accounts to be able to operator against the AWS API. Most of the customers need bounded policy to be security complaint. Here we define a convention to help them to achieve that and a same time have a coherent way to name the roles. From now own all clusters roles we need to create should have the prefix `gs-`. Example:
```
Comment (xh3b4sd) [inline aws/naming_resources.md:5]: Let's please wrap paragraphs by some amount of characters. 80 or 120 works well.

File: aws/naming_resources.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+OLD worker role: `5md7e-worker-EC2-K8S-Role`
+NEW worker role: `gs-5md7e-worker-EC2-K8S-Role`
```
Comment (xh3b4sd) [inline aws/naming_resources.md:8]: Can we write it down schematically next to having examples? 

```
gs-[cluster id]-[what is this?]-EC2-K8S-Role
```

File: aws/naming_resources.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# Naming Resources
+
+## IAM Resources
+
+AWS operator needs a set of roles in customer accounts to be able to operator against the AWS API. Most of the customers need bounded policy to be security complaint. Here we define a convention to help them to achieve that and a same time have a coherent way to name the roles.
```
Comment (rossf7) [inline aws/naming_resources.md:5]:
```suggestion
AWS operator needs a set of roles in customer accounts to be able to act against the AWS API. Most of the customers need a bounded policy to be security complaint. Here we define a convention to help them to achieve this and at the same time have a coherent way to name the roles.
```

File: aws/naming_resources.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+From now own all clusters roles we need to create should have this convention
```
Comment (rossf7) [inline aws/naming_resources.md:7]:
```suggestion
From now on all clusters roles we need to create should have this convention
```

File: aws/naming_resources.md (hunk 3 above)
Comment (xh3b4sd) [inline aws/naming_resources.md:5]: The text is still not wrapped. Let me know when you need help with that.

File: aws/naming_resources.md (hunk 2 above)
Comment (xh3b4sd) [reply aws/naming_resources.md:16]: I would really like to see that scheme in there. Three people liked it apparently so let's get it in as is.

**Comment by rossf7** [comment]
Meant to say LGTM when @xh3b4sd is happy. Just did a quick drive-by.

File: aws/naming_resources.md (hunk 2 above)
Comment (pipo02mix) [reply aws/naming_resources.md:16]: But this is the schema, right?
https://github.com/giantswarm/fmt/pull/105/files#diff-1d117290cfad3e20761fe775d42fff1aR9-R11

File: aws/naming_resources.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
+# Naming Resources
+
+## IAM Resources
+
+AWS operator needs a set of roles in customer accounts to be able to act against the AWS API. Most of the customers need a bounded policy to be security complaint. Here we define a convention to help them to achieve this and at the same time have a coherent way to name the roles.
+
+From now on all clusters roles we need to create should have this convention
+
+- Common prefix `gs-`
+- Follow with `cluster id`
```
Comment (pipo02mix) [inline aws/naming_resources.md:10]:
```suggestion
- Follow with `-cluster_id-`
```

File: aws/naming_resources.md [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+- Ending with `purpose`
```
Comment (pipo02mix) [inline aws/naming_resources.md:11]:
```suggestion
- Ending with `-purpose`
```

File: aws/naming_resources.md [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+From now on all clusters roles we need to create should have this convention
+
+- Common prefix `gs-`
+- Follow with `-cluster_id-`
+- Ending with `-purpose`
+
+Example:
+
+OLD worker role: `5md7e-worker-EC2-K8S-Role`
+NEW worker role: `gs-5md7e-worker-EC2-K8S-Role`
```
Comment (pipo02mix) [inline aws/naming_resources.md:16]:
```suggestion
NEW worker role: `gs-5md7e-master-ec2-k8s` (  gs        5md7e      master-ec2-k8s   )
                                            prefix   cluster_id      purpose
```

File: aws/naming_resources.md [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
+From now on all clusters roles we need to create should have this convention
+
+- Common prefix `gs-`
+- Follow with `-cluster_id-`
+- Ending with `-purpose`
+
+Example:
+
+OLD worker role: `5md7e-worker-EC2-K8S-Role`
```
Comment (pipo02mix) [inline aws/naming_resources.md:15]:
```suggestion
OLD worker role: `5md7e-master-EC2-K8S-Role`
```

File: aws/naming_resources.md (hunk 3 above)
Comment (pipo02mix) [reply aws/naming_resources.md]: Hmm according to Visual code it is 
Screen Shot 2020-04-08 at 16 38 05

**Comment by xh3b4sd** [comment]
I personally do not seem to get what I would like to see but the information we need are there so feel free to go ahead with what you think is right. I do not want to stress about minor details further.

File: aws/naming_resources.md (hunk 2 above)
Comment (pipo02mix) [reply aws/naming_resources.md]: ok I made it more explicit, does it better?

File: aws/naming_resources.md (hunk 1 above)
Comment (pipo02mix) [reply aws/naming_resources.md]: done

File: aws/naming_resources.md (hunk 3 above)
Comment (pipo02mix) [reply aws/naming_resources.md]: done

**Comment by pipo02mix** [comment]
@xh3b4sd I make it more explicit, what do you think now?

**Comment by wtty-fool** [comment]
AWS puts some limitations on IAM role naming. Would it be smart to include most important points here, or maybe link to this part of the documentation: https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_iam-limits.html?

For example a 64 character limit is an important one if we want to be able to switch to a role.

**Comment by pipo02mix** [comment]
@wtty-fool done! PTAL

**Comment by xh3b4sd** [comment]
I am out here guys. Your call. Fanks gang. ❤️

**Comment by pipo02mix** [comment]
I am going with this, @xh3b4sd if you think it can be improved please let me know

**Pull request merged**